src/test/java/software/amazon/cloudformation/resource/ResourceTypeSchemaTest.java

```diff
@@ -244,4 +245,20 @@ static JSONObject getEmptyModel() {
         return new JSONObject().put("id", "required.identifier");
     }
 
+    /**
+     * validate that handler permissions are processed and can be retrieved programatically via the schema object
+     */
+
+    @Test
+    public void getHandlerPermissions_validPermissions_shouldReturnHandlerSetMap() {
+        JSONObject resourceDefinition = loadJSON(HANDLERS_SCHEMA_PATH);
+        ResourceTypeSchema schema = ResourceTypeSchema.load(resourceDefinition);
+
+        assertThat(schema.getHandlerPermissions().get("create")).contains("test:permission");
```

Comment:
I don't like the way to expose internal data as interface, espeically the Map. Others could protential modify it such as `schema.getHandlerPermissions().set("create", new HashSet<>())`.

I would suggest we should expose it as public method, such as 

```
public Set<String> getHandlerPermissions(final String handlerName) {
  return this.schema.getHandlerPermissions().get("create").getPermissions();
}

```